### ISSUE TITLE
Soundness bug QF_NIA

### ISSUE THREAD

**Issue by wintered** [Z3Prover/z3#4122]
```
[559] % z3-4.8.7 small.smt2
sat
sat
[560] % z3 small.smt2
sat
unsat
[561] % 
[561] % cat small.smt2
(declare-fun a () Int)
(declare-fun b () Int)
(declare-fun c () Int)
(declare-fun d () Int)
(declare-fun e () Int)
(declare-fun aa () Int)
(assert (<= 0 a 200))
(assert (>= c 0))
(assert (> e 1))
(assert (= (* c b) e))
(assert (= (+ (* c (- 1)) (* d a)) (* aa (- 120)) a))
(check-sat)
(check-sat-using qfufbv)
[562] 
```

OS: Ubuntu 18.04
Commit: a0de244

**Comment by NikolajBjorner** [comment]
Simpler repro:

```
(declare-const a Int)
(declare-const d Int)
(declare-const b Int)

(assert (<= a 0))
(assert (>= a -1))
(assert (>= (+ (* 120 a) (* (- 120) a d)) 0))
(assert (not (<= (* b (+ (* 120 a) (* (- 120) a d))) 1)))
(check-sat-using smt)
(exit)
``

**Comment by levnach** [comment]
Fixed in 7cfd16c7f948aad0cfb3707695c62ac8f3d3b5fc
